src/ML/src/MANNAutoregressive.cpp

```diff
@@ -249,11 +249,12 @@ bool MANNAutoregressive::initialize(
 }
 
 bool MANNAutoregressive::reset(const MANNInput& input,
-                                     const Contacts::EstimatedContact& leftFoot,
+                                const Contacts::EstimatedContact& leftFoot,
                                      const Contacts::EstimatedContact& rightFoot,
                                      const manif::SE3d& basePosition,
                                      const manif::SE3Tangentd& baseVelocity)
 {
+    // TODO remove 51
     AutoregressiveState state;
     state.pastProjectedBasePositions = std::deque<Eigen::Vector2d>{51, Eigen::Vector2d{0.0, 0.0}};
     state.pastProjectedBaseVelocity = std::deque<Eigen::Vector2d>{51, Eigen::Vector2d{0.0, 0.0}};
```

Comment:
@paolo-viceconte 
![image](https://github.com/ami-iit/bipedal-locomotion-framework/assets/16744101/4385f027-b679-4ef5-b3ba-71abae3542e0)

Reply:
51 is the length of 1 second of past trajectory stored in the autoregressive state. Since the original mocap data are collected at 50 Hz, and therefore the trajectory generation is assumed to proceed at 50 Hz, we need 50 datapoints to store the past second of trajectory. Along with the present datapoint, they sum up to 51!

Reply:
This [comment](https://github.com/ami-iit/bipedal-locomotion-framework/pull/668#discussion_r1204304751) will become a comment of the code

Reply:
Done here bb33760084b1d1a432b7268adf66ea439e07e1e3

Reply:
Is it worth having it as an input parameter?  Ideally, it should be one metadata of the loaded model

Reply:
Yes, I agree. In the next PR, we are going to generate the trajectory from the joypad. I think we can try to handle this in that PR